services/validateAPI/main.py

```diff
@@ -0,0 +1,226 @@
+            os.close(fd)
+            metadata_path = Path(metadata_path_str)
+            with metadata_path.open('w', encoding='utf-8') as f:
+                f.write(metadata_text)
+            logger.info("Metadata CSV created from edited text")
+        elif metadata_csv:
+            metadata_path = save_upload_to_tempfile(metadata_csv)
+            logger.info("Metadata CSV uploaded: %s", metadata_csv.filename)
+        else:
+            logger.warning("No metadata CSV provided")
+            return JSONResponse(status_code=400,
+                                content={
+                                    "ok": False,
+                                    "error": "No metadata CSV provided"
+                                    }
+                                )
+        tmp_files.append(metadata_path)
+
+        query_path = save_upload_to_tempfile(query_fasta)
+        tmp_files.append(query_path)
+        logger.info(
+            "Files prepared | metadata=%s | fasta=%s",
+            metadata_path.name,
+            query_path.name,
+        )
+
+        logger.info("Running p0 validation")
+        rc, out, err = run_p0_validation(
+            VALIDATION_SCRIPT,
+            metadata_path,
+            query_path
+            )
+        logger.info("p0 validation finished | rc=%s", rc)
+        if rc == 0:
+            logger.info("Validation passed successfully")
+            with open(metadata_path, "r", encoding="utf-8") as f:
+                csv_text = f.read()
+            with open(query_path, "r", encoding="utf-8") as f:
+                fasta_text = f.read()
+            return {
+                "ok": True,
+                "message": "Validation passed",
+                "metadata_csv": csv_text,
+                "query_fasta": fasta_text
+            }
+        parsed = parse_errors(err)
+        logger.warning(
+            "Validation failed | type=%s | message=%s",
+            parsed.get("type"),
+            parsed.get("message"),
+        )
+
+        if parsed.get("type") == "metadata_missing_sample":
+            bad_sample_id = parsed.get("sample_id")
+            logger.warning(
+                "Missing sample_id detected | sample_id=%s",
+                bad_sample_id,
+            )
+
+            # if bad_sample_id and " " in bad_sample_id:
+            if bad_sample_id:
+                fixed_sample_id = fix_sample_id_spaces(
+                    metadata_path,
+                    query_path,
+                    bad_sample_id
+                )
+                logger.info(
+                    "Auto-fixed sample_id | old=%s | new=%s",
+                    bad_sample_id,
+                    fixed_sample_id,
+                )
+
+                # re-run validation after auto-fix
+                logger.info("Re-running validation after sample_id fix")
+                rc2, out2, err2 = run_p0_validation(
+                    VALIDATION_SCRIPT,
```

Comment:
I don't think you should need to pass `VALIDATION_SCRIPT` to this if it's calling it as a module now?